### PR TITLE
Set ssh key algorithm for Git Provider Config

### PR DESCRIPTION
**Pull request by mitalipaygude**
*Issue #, if available:*

*Description of changes:*
These code changes let the users set the ssh key algorithm field in the cluster spec.
The new cluster spec would look like:
```
apiVersion: anywhere.eks.amazonaws.com/v1alpha1
kind: FluxConfig
metadata:
  name: fc-byog
spec:
  git:
    repositoryUrl: ssh://git@github.com/owner/byog.git
    sshKeyAlgorithm: ed25519
```

*Testing (if applicable):*
- Unit tests
- Cluster creation
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: pkg/api/v1alpha1/fluxconfig.go [hunk 1]
```diff
@@ -4,10 +4,15 @@ import (
 	"errors"
 	"fmt"
 	"net/url"
+
+	"github.com/aws/eks-anywhere/pkg/logger"
 )
 
 const (
```
Comment (jiayiwang7) [inline pkg/api/v1alpha1/fluxconfig.go:11]:
```suggestion

const (
	Rsa   SshKeyAlgorithmType = "rsa"
	Ecdsa  SshKeyAlgorithmType = "ecdsa"
	Ed25519 SshKeyAlgorithmType = "ed25519"
)

type SshKeyAlgorithmType string
```

File: pkg/api/v1alpha1/fluxconfig.go [hunk 2]
```diff
@@ -70,7 +75,14 @@ func validateGitProviderConfig(config GitProviderConfig) error {
 	if len(config.RepositoryUrl) <= 0 {
 		return errors.New("'repositoryUrl' is not set or empty in gitProviderConfig; repositoryUrl is a required field")
 	}
-
+	if len(config.SshKeyAlgorithm) > 0 {
```
Comment (jiayiwang7) [inline pkg/api/v1alpha1/fluxconfig.go:78]:
```suggestion
	if len(config.SshKeyAlgorithm) > 0 {
		if err := validateSshKeyAlgorithm(config.SshKeyAlgorithm); err != nil {
			return err
		}

	logger.Info("Warning: 'sshKeyAlgorithm' is not set, defaulting to 'ecdsa'")
	
```

File: pkg/executables/flux.go [hunk 3]
```diff
@@ -39,7 +39,7 @@ func (f *Flux) BootstrapToolkitsComponentsGithub(ctx context.Context, cluster *t
 		"--repository", c.Github.Repository,
 		"--owner", c.Github.Owner,
 		"--path", c.ClusterConfigPath,
-		"--ssh-key-algorithm", privateKeyAlgorithm,
+		"--ssh-key-algorithm", defaultPrivateKeyAlgorithm,
```
Comment (jiayiwang7) [inline pkg/executables/flux.go:42]: so for bootstrap github we always use "ecdsa", and for bootstrap git we let user choose ssh algo?

File: pkg/validations/upgradevalidations/immutableFields.go [hunk 4]
```diff
@@ -75,6 +75,9 @@ func ValidateImmutableFields(ctx context.Context, k validations.KubectlClient, c
 				if prevGitOps.Spec.Git.RepositoryUrl != spec.FluxConfig.Spec.Git.RepositoryUrl {
 					return fmt.Errorf("fluxConfig spec.fluxConfig.spec.git.repositoryUrl is immutable")
 				}
+				if prevGitOps.Spec.Git.SshKeyAlgorithm != spec.FluxConfig.Spec.Git.SshKeyAlgorithm {
```
Comment (jiayiwang7) [inline pkg/validations/upgradevalidations/immutableFields.go:78]: can we pls unit test this?

**Review by jiayiwang7 — COMMENTED** [review]
Looks good!

File: pkg/executables/flux.go (hunk 3 above)
Comment (mitalipaygude) [reply pkg/executables/flux.go:42]: based on documentation here https://fluxcd.io/docs/cmd/flux_bootstrap_github/
both github and git support all 3 algorithms

File: pkg/api/v1alpha1/fluxconfig.go (hunk 1 above)
Comment (mitalipaygude) [reply pkg/api/v1alpha1/fluxconfig.go:11]: will follow up on this for sure!

Comment (mitalipaygude) [reply pkg/api/v1alpha1/fluxconfig.go:11]: https://github.com/aws/eks-anywhere/issues/2036

File: pkg/executables/flux.go (hunk 3 above)
Comment (mitalipaygude) [reply pkg/executables/flux.go:42]: https://github.com/aws/eks-anywhere/issues/2037

**Pull request closed without merging**